### DISCUSSION TITLE
How to change  DataGridCellStyle?

### DISCUSSION THREAD

**Discussion by ApaulMyLittleAirport** [AvaloniaUI/Avalonia#8362] [Q&A] [answered]
I want the border width to be 0 when the mouse selects the cell，But it didn't work.
```		
<Style Selector="DataGridCell:selected /template/ Border">
			<Setter Property="BorderThickness" Value="0"/>
		</Style>
```

**Comment by maxkatz6** [comment, marked as answer]
You can use DevTools to investigate what's actual value is set, are pseudoclasses correct as well as the selector. It also will show list of values assigned to the property.
https://docs.avaloniaui.net/docs/getting-started/developer-tools
And this might help https://docs.avaloniaui.net/docs/styling/troubleshooting

**Reply by donnyv** [reply]
My dude, these dev tools are the bomb 👍

**Reply by ApaulMyLittleAirport** [reply]
> 我的伙计，这些开发工具是炸弹👍



> 您可以使用 DevTools 来调查实际值的设置、伪类是否正确以及选择器。它还将显示分配给属性的值列表。 https://docs.avaloniaui.net/docs/getting-started/developer-tools 这可能对 https://docs.avaloniaui.net/docs/styling/troubleshooting 有所帮助

Thanks for providing this awesome tool, I'll give it a try!

**Comment by dexchix** [comment]
I found the problem. This is not related to the border, but to the rectangle.  
Add Style

```
<Style Selector="DataGridCell Rectangle">
					<Setter Property="IsVisible" Value="False"></Setter>
				</Style>
```

**Comment by Immortal-Sty** [comment]
Use this:

```xml
<!-- Remove the highlight when line selected. -->
<Style Selector="DataGridRow:selected Rectangle#BackgroundRectangle">
  <Setter Property="IsVisible" Value="False" />
</Style>

<!-- Remove the border when cell focused. -->
<Style Selector="DataGridCell:focus Grid#FocusVisual">
  <Setter Property="IsVisible" Value="False" />
</Style>
```

Thanks for [DeepWiki](https://deepwiki.com/AvaloniaUI/Avalonia.Controls.DataGrid).
